File: apps/backend/src/EnterpriseAgentOs.Domain/features/Agents/IAgentDeployer.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public sealed record AgentDeployment(string PodName, string ServiceUrl);

public interface IAgentDeployer
{
    Task<AgentDeployment> DeployAsync(
        Guid agentId,
        CancellationToken ct = default);

    Task<bool> RemoveAsync(string podName, CancellationToken ct = default);

    Task<string> GetStatusAsync(string podName, CancellationToken ct = default);

    Task<string> GetLogsAsync(string podName, int tailLines = 200, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentDeployer.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.concurrent.CompletableFuture;

public interface IAgentDeployer {
    CompletableFuture<Boolean> removeAsync(String podName);
    CompletableFuture<String> getStatusAsync(String podName);
    CompletableFuture<String> getLogsAsync(String podName, int tailLines);
}
